### DISCUSSION TITLE
How to refresh the list after a bulk action?

### DISCUSSION THREAD

**Discussion by spekulatius** [rappasoft/laravel-livewire-tables#263] [Q&A]
Hey,

Thanks for building this great package. I'm just implementing in my first project. I was wondering how I trigger a refresh after a bulk action? I'm deleting records. After the records are delete the selection information and the "bulk action menu" remain open:

![image](https://user-images.githubusercontent.com/8433587/116817460-81149600-ab77-11eb-952a-fbfd9355d8b5.png)

How would trigger a refresh/rerendering after the bulk action?

Cheers,
Peter

**Comment by rappasoft** [comment]
Hmm that's a good question i've only ever tested with exports. You could try emitting this event at the end of your bulk method and see what happens: https://github.com/rappasoft/laravel-livewire-tables/blob/master/src/DataTableComponent.php#L90

Otherwise I'll have to think of something else.

**Reply by spekulatius** [reply]
Hey Anthony,

Thank you for the hint, but it didn't solve the issue. The information about the selection and the bulk-menu remain as before.

Peter

**Reply by spekulatius** [reply]
I've checked the code a little and found the `resetBulk` method. It doesn't close the bulk menu but resets the counters. That should work for now. I would have updated the wiki but I didn't find an option to do it.

**Reply by rappasoft** [reply]
I was just about to suggest adding:

```php
$this->resetAll();
$this->filters['search'] = null;
```

To the end of the bulk, it will essentially set it back to page load but you'll lose any searching you did to get to the results you had. There are other reset methods for specific things you can use to get a more specific result, you can also set individual properties because you have access to them on your component. I.e. `$this->sorts $this->filters`, etc. I will add a better method to reset everything and add it to the docs.

**Reply by spekulatius** [reply]
Yeah, I was thinking about resetting stuff myself but opted against it if I find a different solution (a method call). Thanks for helping so fast :)

**Reply by rappasoft** [reply]
I made a patch, 1.5.1 that beefed up the method, and a [wiki](https://github.com/rappasoft/laravel-livewire-tables/wiki/Resetting-after-bulk-actions) you can use.

**Reply by spekulatius** [reply]
Awesome, I'll upgrade shortly :)
